File: CSA_HW2/Fish.cpp
#include "Fish.h"

//------------------------------------------------------------------------------
// -------------------- CPP FILE OF CHILD CLASS 'FISH' -------------------------
//------------------------------------------------------------------------------

// Implementation of the method for printing information about fish.
void Fish::Output(std::ofstream &output_file_stream) {
    output_file_stream << "Animal " << name_ << " with weight " << weight_ << " grams. \n";
    output_file_stream << "It's quotient from name to weight is " << Quotient() << ". \n";
    output_file_stream << "It's a fish, and it lives in the ";
    switch (habitat_) {
        case 1: output_file_stream << "river. \n";
            break;
        case 2: output_file_stream << "sea. \n";
            break;
        case 3: output_file_stream << "lake. \n";
            break;
        case 4: output_file_stream << "ocean. \n";
            break;
        default: output_file_stream << "pond. \n";
    }
}

File: CSA_HW3/Animals/Fish.py
from Animals.Animal import Animal


# ------------------------------------------------------------------------------
# --------------------------- CHILD CLASS 'FISH' -------------------------------
# ------------------------------------------------------------------------------

class Fish(Animal):
    habitat_types = ["river", "sea", "lake", "ocean", "pond"]

    # Constructor for child class fish with extra parameter - fish habitat.
    def __init__(self, name, weight, habitat):
        super(Fish, self).__init__(name, weight)
        self.__habitat__ = habitat

    # Method for printing information about fish.
    def out(self):
        # Calling method for print basic info about fish.
        info = super(Fish, self).out()
        info += "It's a fish, and it lives in the "
        if self.__habitat__ < 1 or self.__habitat__ > 5:
            print("Fish habitat type was incorrect so it was changed to default value (pond)")
            self.__habitat__ = 5
        info += self.habitat_types[self.__habitat__ - 1] + ".\n"
        return info
